Fix the regex for URL parser

nirmit opened the pull request:
Missing escapes resulted in incorrect regex being generated that caused issue #147.  Tested via decaffienated.

Fixes #147

seesemichaelj commented:
FYI, the conflicted files are because you based your `urlfix-147` version on your repo's version of `master` which is out of date from our master. you should rebase your `master` branch before creating a new feature branch to prevent conflicts

seesemichaelj commented:
I'll look at this tomorrow or Thursday; I don't think I agree with every change here within #145 and/or #148. Definitely earmarked some stuff in here that may or may not have needed to be in to fix the `?` issue in #143.

nirmit commented:
@seesemichaelj and @dweremeichik 
This uses a modified version of @diegoperini's regex (https://mathiasbynens.be/demo/url-regex).  It passes almost all the test cases.

1.  Modified it to handle URLs without a scheme. (www.google.com)
2.  Added ability to handle `mailto` and `tel`
3.  Added support for punycode URLs.
